src/main/kotlin/uk/gov/justice/digital/hmpps/communitypaybackapi/common/service/OffenderService.kt

```diff
@@ -0,0 +1,69 @@
+package uk.gov.justice.digital.hmpps.communitypaybackapi.common.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.communitypaybackapi.common.client.CaseAccess
+import uk.gov.justice.digital.hmpps.communitypaybackapi.common.client.CaseSummary
+import uk.gov.justice.digital.hmpps.communitypaybackapi.common.client.CommunityPaybackAndDeliusClient
+
+@Service
+class OffenderService(
+  val communityPaybackAndDeliusClient: CommunityPaybackAndDeliusClient,
+  val contextService: ContextService,
+) {
+
+  companion object {
+    const val MAX_OFFENDER_REQUEST_COUNT = 500
+  }
+
+  fun getOffenderInfo(
+    crns: Set<String>,
+  ): List<OffenderInfoResult> {
+    require(crns.size <= MAX_OFFENDER_REQUEST_COUNT) { "Can only request up-to $MAX_OFFENDER_REQUEST_COUNT CRNs. Have requested ${crns.size}." }
+
+    if (crns.isEmpty()) {
+      return emptyList()
+    }
+
+    val caseSummaryByCrn = communityPaybackAndDeliusClient.getCaseSummaries(crns)
+      .cases.associateBy(
+        keySelector = { it.crn },
+        valueTransform = { it },
+      )
+
+    val laoCrns = crns.filter { caseSummaryByCrn[it]?.hasLimitations() == true }.toSet()
+
+    val userAccessByCrn = if (laoCrns.isNotEmpty()) {
+      communityPaybackAndDeliusClient.getUsersAccess(contextService.getUserName()!!, laoCrns)
```

Comment:
Could we handle there not being a user name within getUserName() itself, and throw an exception "user name not found"?

Reply:
Rather than !! here